### ISSUE TITLE
Bun install does not install dependencies between packages in the same monorepo

### ISSUE THREAD

**Issue by nbbaier** [oven-sh/bun#4650]
### What version of Bun is running?

1.0.0+822a00c4d508b54f650933a73ca5f4a3af9a7983

### What platform is your computer?

Darwin 22.5.0 arm64 armw

### What steps can reproduce the bug?

Following the guide for setting up a monorepo [here](https://bun.sh/guides/install/workspaces):

Project structure:
```
.
├── package.json
├── node_modules
└── packages
    ├── stuff-a
    │   └── package.json
    └── stuff-b
        └── package.json
```

Root `package.json`
```
{
  "name": "my-monorepo",
  "private": true,
  "workspaces": ["packages/*"]
}
```

Stuff-b `package.json`
```
{
  "name": "stuff-b",
  "dependencies": {
   "stuff-a": "*"
  }
}
```

### What is the expected behavior?

Running `bun install` from the project root to install dependencies for all workspaces.

### What do you see instead?

I get the following error:

```
bun install v1.0.0 (822a00c4)
  🔍 stuff-a [1/1] 
error: package "stuff-a" not found registry.npmjs.org/stuff-a 404
error: stuff-a@* failed to resolve
```

**Comment by divmgl** [comment]
You need to provide a version for `stuff-a`. `*` won't work.

**Comment by nbbaier** [comment]
Thanks, but even with an explicit version, it still fails.

Revised `packages/stuff-b/package.json`
```
{
  "name": "stuff-b",
  "dependencies": {
    "stuff-a": "1.0.0"
  }
}
```

`packages/stuff-a/package.json`

```
{
    "name": "stuff-a",
    "version": "1.0.0"
}
```
Error:
```
bun install v1.0.0 (822a00c4)
  🔍 stuff-a [1/1] 
error: package "stuff-a" not found registry.npmjs.org/stuff-a 404
error: stuff-a@1.0.0 failed to resolve
```

**Comment by nil1511** [comment]
Faced a similar issue. explicitly specifying as workspace dependency fixes it
```{
  "name": "stuff-b",
  "dependencies": {
   "stuff-a": "workspace:*"
  }
}
```

**Comment by ramiel** [comment]
This didn't fix the problem for me

**Comment by noClaps** [comment]
I've found that you don't need to declare packages in a monorepo as dependencies of each other. They're automatically added to `node_modules`, so you can use them directly.

```diff
"dependencies": {
// other dependencies
-  "stuff-a": "*"
}
```

**Comment by rgripper** [comment]
Removing workspace dependency didn't help in my case

**Comment by benvan** [comment]
@ramiel I had a similar issue. Re. "This didn't fix the problem for me", try deleting your bun lock file and re-running bun install
Context https://github.com/oven-sh/bun/issues/4676

**Comment by nbbaier** [comment]
@benvan that worked for a first `bun install` but then if I run `bun add zod` (or something else) in `packages/stuff-b`, I get the same error, i.e. the `stuff-a` dependency can't be resolved and the add fails

**Comment by 2coo** [comment]
I am facing the [same behavior](https://github.com/oven-sh/bun/issues/4650#issuecomment-1712571131) as his

**Comment by ramiel** [comment]
Removing the lock file worked for me

**Comment by dylan-conway** [comment]
This is fixed in bun `v1.1.8` and newer versions.

> @benvan that worked for a first `bun install` but then if I run `bun add zod` (or something else) in `packages/stuff-b`, I get the same error, i.e. the `stuff-a` dependency can't be resolved and the add fails

We also landed #11177 in `v1.1.9` which fixes adding dependencies within a workspace.

Please reopen or create a new issue if you continue to find install bugs in the latest release of bun.
